WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -173,7 +182,9 @@ private void unpackPostIntent(@NonNull Intent intent) {
             if (!hasPendingOrInProgressMediaUploadsForPost(post)) {
                 mPostUploadHandler.upload(post);
             } else {
-                sPostsWithPendingMedia.add(post);
+                List<MediaModel> pendingMedia = MediaUploadHandler.getPendingOrInProgressMediaUploadsForPost(post);
+                UploadingPost uploadingPost = new UploadingPost(post, pendingMedia);
```

Comment:
There is a problem with this: as we always calculate stuff starting from whatever `pendingMedia` consists of at any given time, if you tap on the post to edit it, wait until one or more media items finish upload while you're in the editor, and then come back to the posts list, the progress bar looks different each time, because the calculation is proportional to _ all pending things at this moment_  as opposed to  _all pending things since I started uploading everything in this post_.

This happens this way because we stop/cancel the UploadService each time we try to Edit, and so the service is triggered with a new Intent.

I suggest (if possible) that we use the full count of media items inside a Post to calculate progress when the UploadService receives a new intent, not just the current pending items.